tests/spread/general/big/task.yaml

```diff
@@ -50,3 +50,8 @@ execute: |
   docker cp big-container:/usr/bin/pebble - | tar xf -
   file pebble | grep "statically linked"
   docker rm -f big-container
+
+  ############################################################################################
+  # check that the /bin symlink to /usr/bin wasn't broken by the /bin/new_bin_file addition
+  ############################################################################################
+  docker run --rm big ls /bin/bash /bin/new_bin_file
```

Comment:
isn't this file always going to be there? I guess you should check instead that this **isn't** the only file, and that all other binaries are symlinks instead of just this one...?

Reply:
If I understand correctly, the files in /usr/bin won't appear as symlinks here.

The best way to test this is probably to put a file into /usr/bin and then check for the existence of that file in /bin

Reply:
Hmm I'm not sure I see the issue here. The command `ls /bin/bash /bin/new_bin_file` will only succeed if both `/bin/bash` and `/bin/new_bin_file` exist:

```bash
$ ls /bin/bash /bin/sbrubbles
ls: cannot access '/bin/sbrubbles': No such file or directory
/bin/bash
$ echo $?
2
```

Since `bash` is actually `/usr/bin/bash`, the command will only succeed if the `/bin` symlink to `/usr/bin` was successfully preserved, right? Note that the files themselves aren't symlinks, only `/bin` is.

Reply:
Are we guaranteed that the bash file will be `/usr/bin/bash` in future? (E.g. might that get swapped around and `/usr/bin` symlinks to `/bin` at some point?) 

Controlling both files being tested might be more robust to unexpected changes upstream, but it may also be overkill.

Reply:
Yes indeed that's a more coherent test. In fact, any of the following tests seems sane enough:
 - len(/usr/bin) = len(/bin)+1
 - /bin -> /usr/bin
 - [ls /bin] = [ls /usr/bin , /bin/new_bin_file]

Reply:
wait but the first and third checks aren't possible right? Since `/bin` is just a symlink to `/usr/bin`, there is no way for something to be reachable via `/bin` but not via `/usr/bin`

Reply:
yes you're right...it doesn't work as is. It would only make (some) sense if you could compare before and after adding the new_bin_file.

nevermind ;)